src/freenet/clients/http/WelcomeToadlet.java

```diff
@@ -80,7 +88,42 @@ private void addCategoryToList(BookmarkCategory cat, HTMLNode list, boolean noAc
                 HTMLNode cell = row.addChild("td", "style", "border: none;");
                 if (item.hasAnActivelink() && !noActiveLinks) {
                     String initialKey = item.getKey();
-                    String key = '/' + initialKey + (initialKey.endsWith("/") ? "" : "/") + "activelink.png";
+                    // extract the key type
+                    String keyType = initialKey.substring(1,3);
+                    String key = '/' + initialKey + (initialKey.endsWith("/") ? "" : "/");
+					Matcher match = null;
+                    switch (keyType) {
+                        case "USK":
+                        case "SSK": 
+                            match = PATTERN_USK_SSK.matcher(key);
```

Comment:
this only matches when the key is already well-formed ⇒ ideally extract to a static method (uses no `this`; just put `static` before the function name to compiler-check that), and write a test that takes the existing bookmarks as input (and maybe some bad but valid examples) and returns the correct key for the activelink.